paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -1070,9 +1070,13 @@ internal class CustomerSheetViewModel(
                 }?.let {
                     PaymentSelection.Saved(it)
                 } ?: state.currentSelection
-
+                val filteredPaymentMethods = paymentMethods.filter { paymentMethod ->
```

Comment:
Should the filter be happening in `retrievePaymentMethods` instead?

Reply:
I did look at that approach and is also valid, but it became tricky (at least for me) given that it returns some kind of "task".  IMO both places seem reasonable.

```kotlin
    override suspend fun retrievePaymentMethods() = runCatchingAdapterTask {
        customerAdapter.retrievePaymentMethods()
    }
```